### ISSUE TITLE
[Paper Note] Unsupervised Pretraining for Sequence to Sequence Learning, Prajit Ramachandran+, EMNLP'17, 2016.11

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#83]
# URL
- preprint: https://arxiv.org/abs/1611.02683
- acl anthology:http://aclweb.org/anthology/D17-1039
# Authors
  - Prajit Ramachandran
  - Peter J. Liu
  - Quoc V. Le
# Abstract
  - This work presents a general unsupervised learning method to improve the accuracy of sequence to sequence (seq2seq) models. In our method, the weights of the encoder and decoder of a seq2seq model are initialized with the pretrained weights of two language models and then fine-tuned with labeled data. We apply this method to challenging benchmarks in machine translation and abstractive summarization and find that it significantly improves the subsequent supervised models. Our main result is that pretraining improves the generalization of seq2seq models. We achieve state-of-the art results on the WMT English$\rightarrow$German task, surpassing a range of methods using both phrase-based machine translation and neural machine translation. Our method achieves a significant improvement of 1.3 BLEU from the previous best models on both WMT'14 and WMT'15 English$\rightarrow$German. We also conduct human evaluations on abstractive summarization and find that our method outperforms a purely supervised learning baseline in a statistically significant manner.
# Translation (by gpt-4o-mini)
- 本研究では、シーケンスからシーケンス（seq2seq）モデルの精度を向上させるための一般的な教師なし学習手法を提案します。私たちの手法では、seq2seqモデルのエンコーダとデコーダの重みを2つの言語モデルの事前学習済みの重みで初期化し、その後ラベル付きデータでファインチューニングを行います。この手法を機械翻訳や抽象的要約の難しいベンチマークに適用したところ、後続の教師ありモデルの性能が大幅に向上することがわかりました。私たちの主な結果は、事前学習がseq2seqモデルの一般化能力を向上させるということです。WMT英語→ドイツ語タスクにおいて最先端の結果を達成し、フレーズベースの機械翻訳とニューラル機械翻訳の両方を使用したさまざまな手法を上回りました。私たちの手法は、WMT'14およびWMT'15英語→ドイツ語の両方で、以前の最良モデルから1.3 BLEUの大幅な改善を達成しました。また、抽象的要約に関する人間評価を実施したところ、私たちの手法は純粋な教師あり学習のベースラインを統計的に有意に上回ることがわかりました。
# Summary (by gpt-4o-mini)
- 本研究では、seq2seqモデルの精度向上のために、事前学習済みの言語モデルの重みでエンコーダとデコーダを初期化し、ラベル付きデータでファインチューニングする教師なし学習手法を提案。機械翻訳や抽象的要約のベンチマークで性能が大幅に向上し、特にWMT英語→ドイツ語タスクで最先端の結果を達成。BLEUスコアで1.3の改善を記録し、抽象的要約においても教師あり学習のベースラインを有意に上回った。

**Comment by AkihikoWatanabe** [comment]
seq2seqにおいてweightのpretrainingを行う手法を提案
seq2seqでは訓練データが小さいとoverfittingしやすいという弱点があるので、大規模なデータでunsupervisedにpretrainingし、その後目的のデータでfinetuneすることで精度を向上させましょう、というお話。
WMTの翻訳タスクにおいて、1.3ポイント BLEUスコアが改善、abstractive summarizationでも実験したが、精度は向上せず。しかしながら要約ではpretrainingによってrepetitionが減少したと主張。

encoder, decoderそれぞれを切り離して考えると、それぞれ言語モデルとみなすことができるため(encoderにはoutput-layerを追加)、それぞれの言語モデルを独立に大規模なラベルなしデータでpretrainingする。
fine-tuneする際は、targetデータだけでなく、pretrainingする際のデータも同時に学習を続ける（LM Objective）
LM Objectiveは、target側のobjective functionにpretraining側のobjective functionの項を重み付きで追加したもの。

Abltion studyによると、MTにおいてはsoftmax-layerをpretrainingすることが重要。softmax-layerのpretrainingをablationするとBLEUスコアが1.6ポイント減少。
LM objectiveをなくすと、pretrainingの効果がほとんどなくなる(BLEUスコア-2.0ポイント)。
sumarizationにおいては、embeddingのpretrainingが大幅なROUGEスコアの改善を見せた。また、MTと異なり、encoder側のpretrainingがスコア向上に寄与。

LM Objectiveは結構使えそうな印象
